### PR TITLE
Add support for CA bundles

### PR DESCRIPTION
**Pull request by moritz89**
Instead of only a single CA cert, support CA bundles to verify the TLS connections.

https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/protocols/esp_crt_bundle.html

To add support, one option is to add a new constructor that accepts a pointer and then calls setCACertBundle on the WiFiClientSecure instance.

Espressif supports these bundles via an archive that reduces the required flash memory and only loads the CA indexes instead of the entire certificate into memory. The respective cert is then loaded on TLS connection setup.

See #884

### CODE REVIEW

File: examples/esp32/WebSocketClientSSLBundle/src/main.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
+#include "main.hpp"
+
```
Comment (Links2004) [inline examples/esp32/WebSocketClientSSLBundle/src/main.cpp:2]: this file does not look like a example for the Client.
may the code of examples/esp32/WebSocketClientSSLBundle/example.cpp needs to be here?

Comment (moritz89) [reply examples/esp32/WebSocketClientSSLBundle/src/main.cpp:7]: I updated the main.cpp to handle the websocket connection. The problem is that for the example to be useful it has to include the `x509_crt_bundle` file which I only know how to link with platformIO. I also added an INO version that does everything except include the bundle.

Comment (moritz89) [reply examples/esp32/WebSocketClientSSLBundle/src/main.cpp:7]: I pushed changes to fix the Arduino build

Comment (moritz89) [reply examples/esp32/WebSocketClientSSLBundle/src/main.cpp:7]: I updated the build with a workaround. I am not deep enough in this build system or the incbin library, but that should be the path forward.

Comment (Links2004) [reply examples/esp32/WebSocketClientSSLBundle/src/main.cpp:7]: looks like that `arduino-cli` tries to build the ino and the main.cpp which results in conflicts.

Comment (moritz89) [reply examples/esp32/WebSocketClientSSLBundle/src/main.cpp]: I separated the the PIO build into a separate folder. That might work. The build works locally now for me

**Comment by Links2004** [comment]
thanks for the contribution

**Pull request merged**